pojo/src/main/java/com/spotify/hamcrest/pojo/IsPojo.java

```diff
@@ -38,9 +47,10 @@
 public class IsPojo<A> extends TypeSafeDiagnosingMatcher<A> {
 
   private final Class<A> cls;
-  private final ImmutableMap<String, Matcher<?>> methodMatchers;
+  private final ImmutableMap<Function<A, ?>, Matcher<?>> methodMatchers;
```

Comment:
I think this might be a bit better if there was a common base classes for the value suppliers, like `ValueSupplier<A, T>`. It could have a virtual method `getName()` or something, and would implement `hashCode()` based on that value. 

Then this map would be `ImmutableMap<ValueSupplier<A, ?>, Matcher<?>>` and have safe hashCodes. In the code with the mismatches, we just need to call `supplier.getName()` or `supplier.toString()`.